### PR TITLE
Address warning in setup-python

### PR DESCRIPTION
**Pull request by isarkis**
Explicitly set python-version to avoid throwing a warning.

b/284376258

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -43,6 +43,8 @@ jobs:
           persist-credentials: false
       - name: Setup Python
         uses: actions/setup-python@v4
+        with:
+          python-version: '^3.8.x'
```
Comment (kaidokert) [inline .github/workflows/lint.yaml:47]: Hmm. I think we were aiming for Python 3.7 as the baseline. Oscar ran into various issues between .7 and .8, we should keep it consistent.

See https://github.com/youtube/cobalt/blob/main/.github/workflows/pytest.yaml#L21

**Review by kaidokert — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (isarkis) [reply .github/workflows/lint.yaml:47]: @andrewsavage1, you recommended 3.8.x, any objections if we go with 3.7.x?

Comment (andrewsavage1) [reply .github/workflows/lint.yaml:47]: That sounds okay

Comment (isarkis) [reply .github/workflows/lint.yaml]: Changed to 3.7.x

**Review by kaidokert — APPROVED** [review]

**Pull request merged**